packages/shared/friends/sagas.ts

```diff
@@ -1928,36 +1928,37 @@ export async function getChannelMembers(request: GetChannelMembersPayload) {
     members: []
   }
 
-  const members = getMembers(client, channel.userIds ?? [], request.channelId)
-    .filter(({ name }) => {
-      const searchTerm = request.userName.toLocaleLowerCase()
-      const lowerCaseName = name.toLocaleLowerCase()
-      return lowerCaseName.search(searchTerm) >= 0
-    })
-    .slice(request.skip, request.skip + request.limit)
+  const allMembers = getMembers(client, channel.userIds ?? [], request.channelId).filter(({ name }) => {
+    const searchTerm = request.userName.toLocaleLowerCase()
+    const lowerCaseName = name.toLocaleLowerCase()
+    return lowerCaseName.search(searchTerm) >= 0
+  })
 
-  if (members.length === 0) {
+  if (allMembers.length === 0) {
     // it means the channel has no members
     getUnityInstance().UpdateChannelMembers(channelMembersPayload)
     return
   }
 
-  const ownId = client.getUserId()
-
-  // update catalog with missing users, by using default profiles with name and image url
-  sendMissingProfiles(members, ownId)
-
   // we only notify members who are online if presence is enabled, else every joined member
-  const memberIds = members.map((member) => member.userId)
+  const memberIds = allMembers.map((member) => member.userId)
   const onlineOrJoinedMemberIds = getOnlineOrJoinedMembers(memberIds, client)
```

Comment:
This is not the best approach especially when there are many members joined, but I thought of it that way because at the moment the explorer team sets the skip value based on how many members they have loaded. We wouldn't be able to know what our last page was, at least we changed the payloads so that we send each other that information. Any thoughts reviewers?